ethcore/wasm/run/src/runner.rs

```diff
@@ -0,0 +1,204 @@
+				write!(
+					f,
+					"Expected to return result: 0x{} ({} bytes), but got 0x{} ({} bytes)",
+					expected.to_hex(),
+					expected.len(),
+					actual.to_hex(),
+					actual.len()
+				),
+			UsedGas { expected, actual } =>
+				write!(f, "Expected to use gas: {}, but got actual gas used: {}", expected, actual),
+
+			Runtime(ref s) =>
+				write!(f, "WASM Runtime error: {}", s),
+
+			Load(ref e) =>
+				write!(f, "Load i/o error: {}", e),
+
+			NoCall(ref call) =>
+				write!(f, "Call not found: {:?}", call),
+
+			StorageMismatch { ref key, ref expected, actual: Some(ref actual)} =>
+				write!(
+					f,
+					"Storage key {} value mismatch, expected {}, got: {}",
+					key.as_ref().to_vec().to_hex(),
+					expected.as_ref().to_vec().to_hex(),
+					actual.as_ref().to_vec().to_hex(),
+				),
+
+			StorageMismatch { ref key, ref expected, actual: None} =>
+				write!(
+					f,
+					"No expected storage value for key {} found, expected {}",
+					key.as_ref().to_vec().to_hex(),
+					expected.as_ref().to_vec().to_hex(),
+				),
+		}
+	}
+}
+
+pub fn run_fixture(fixture: &Fixture) -> Vec<Fail> {
+	let mut params = ActionParams::default();
+
+	params.code = Some(Arc::new(
+		match load_code(fixture.wasm_file.as_ref()) {
+			Ok(code) => code,
+			Err(e) => { return Fail::load(e); },
+		}
+	));
+
+	if let Some(ref address) = fixture.address {
+		params.address = address.clone().into();
+	}
+
+	if let Some(gas_limit) = fixture.gas_limit {
+		params.gas = U256::from(gas_limit);
+	}
+
+	if let Some(ref data) = fixture.payload {
+		params.data = Some(data.clone().into())
+	}
+
+	if let Some(value) = fixture.value {
+		params.value = ActionValue::Transfer(value.clone().into())
+	}
+
+	let mut ext = FakeExt::new();
+	if let Some(ref storage) = fixture.storage {
+		for storage_entry in storage.iter() {
+			let key: U256 = storage_entry.key.into();
+			let val: U256 = storage_entry.value.into();
+			ext.store.insert(key.into(), val.into());
+		}
+	}
+
+	let mut interpreter = wasm_interpreter();
+
+	let interpreter_return = match interpreter.exec(params, &mut ext) {
+		Ok(ret) => ret,
+		Err(e) => { return Fail::runtime(e); }
+	};
+	let (gas_left, result) = match interpreter_return {
+		GasLeft::Known(gas) => { (gas, Vec::new()) },
+		GasLeft::NeedsReturn { gas_left: gas, data: result, apply_state: _apply } => (gas, result.to_vec()),
+	};
+
+	let mut fails = Vec::new();
+
+	for assert in fixture.asserts.iter() {
+		match *assert {
+			Assert::Return(ref data) => {
+				if &data[..] != &result[..] {
+					fails.push(Fail::Return { expected: (&data[..]).to_vec(), actual: (&result[..]).to_vec() })
+				}
+			},
+			Assert::UsedGas(gas) => {
+				let used_gas = fixture.gas_limit.unwrap_or(0) - gas_left.low_u64();
+				if gas != used_gas {
+					fails.push(Fail::UsedGas { expected: gas, actual: used_gas });
+				}
+			},
+			Assert::HasCall(ref locator) => {
+				let mut found = false;
+
+				for fake_call in ext.calls.iter() {
+					let mut mat4 = true;
```

Comment:
What is mat4?

Reply:
`match` is a reserved word.. :)

Reply:
Ah, I see, thanks! 

Maybe it will be a little more legible to use `match_found`? 
Also, AFAIK, in Rust there is a common idiom to use underscore suffix in such cases, i.e. `match_`